src/System.Management.Automation/utils/MetadataExceptions.cs

```diff
@@ -21,6 +20,7 @@ public class MetadataException : RuntimeException
         /// </summary>
         /// <param name="info">Serialization information.</param>
         /// <param name="context">Streaming context.</param>
+        [Obsolete("Legacy serialization support is deprecated since .NET 8", DiagnosticId = "SYSLIB0051")]
         protected MetadataException(SerializationInfo info, StreamingContext context) : base(info, context)
         {
             SetErrorCategory(ErrorCategory.MetadataError);
```

Comment:
If we throw `NotSupportedException` elsewhere, then I think we should do the same here.
Or, if you depend on `base(info, context)` to throw, then you should use the same pattern consistently elsewhere.

Reply:
throwing here too.